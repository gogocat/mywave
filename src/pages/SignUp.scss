.sign-up {
    &__success-message {
        margin-top: 2rem;
        text-align: center;
    }
}