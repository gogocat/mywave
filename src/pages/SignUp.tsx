import { useState } from 'react';
import Avatar from '@mui/material/Avatar';
import Paper from '@mui/material/Paper';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import LockIcon from '@mui/icons-material/Lock';
import Typography from '@mui/material/Typography';
import SignUpForm from 'components/Forms/SignUpForm';
import Tick from 'components/Tick';

import './SignUp.scss';

function SuccessMessage() {
    return (
        <div className="sign-up__success-message">
            <Tick size="lg" />
            <h3 className="text-center">Success!</h3>
            <p>Thank you registered.</p>
        </div>
    );
}

function SignUp() {
    const [isCompleted, setIsCompleted] = useState(false);

    return (
        <Grid 
            container
            className="sign-up"
            component="main"
            sx={{ height: '100vh' }}>
            <Grid
                item
                xs={false}
                sm={4}
                md={7}
                sx={{
                    backgroundImage: 'url(https://source.unsplash.com/random?wallpapers)',
                    backgroundRepeat: 'no-repeat',
                    backgroundColor: (t) =>
                        t.palette.mode === 'light' ? t.palette.grey[50] : t.palette.grey[900],
                    backgroundSize: 'cover',
                    backgroundPosition: 'center',
                }}
            />
            <Grid item
                xs={12}
                sm={8}
                md={5}
                component={Paper}
                elevation={6}
                square
            >
                <Box
                    sx={{
                        my: 8,
                        mx: 4,
                        display: 'flex',
                        flexDirection: 'column',
                        alignItems: 'center',
                    }}
                >
                    <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
                        <LockIcon />
                    </Avatar>
                    <Typography component="h1"
                        variant="h5">
                        Sign Up
                    </Typography>
                    { isCompleted 
                        ? <SuccessMessage />
                        : <SignUpForm onCompleted={setIsCompleted}/>
                    }
                </Box>
            </Grid>
        </Grid>
    );
}

export default SignUp;
