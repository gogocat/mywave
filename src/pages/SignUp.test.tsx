import {
    render, 
    screen 
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SignUp from 'pages/SignUp';

describe('Sing Up screen', () => {
    it('should render signup form', ()=>{
        const {
            asFragment
        } = render(<SignUp />);

        const heading = screen.getByText('Sign Up', { selector: 'h1' });
        const successHeading = screen.queryByText('Success!', { selector: 'h3' });

        expect(heading).toBeInTheDocument();
        expect(successHeading).not.toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it('should submit signup form', async()=> {
        const user = userEvent.setup();

        const {
            asFragment
        } = render(<SignUp />);

        const VALID_VALUES = {
            userName: 'test@test.com',
            password: 'A1#',
        };

        const userNameField: HTMLInputElement = screen.getByLabelText('Email Address');
        const passwordField: HTMLInputElement = screen.getByLabelText('Password');
        const confirmPasswordField: HTMLInputElement = screen.getByLabelText('Confirm password');
        const submitButton = screen.getByText('Sign Up', { selector: 'button' });

        await user.type(userNameField, VALID_VALUES.userName);
        await user.type(passwordField, VALID_VALUES.password);
        await user.type(confirmPasswordField, VALID_VALUES.password);
        await userEvent.click(submitButton);

        const successHeading = await screen.findByText('Success!', { selector: 'h3' });
        
        expect(successHeading).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});