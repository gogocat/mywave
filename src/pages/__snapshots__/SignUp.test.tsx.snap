// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sing Up screen should render signup form 1`] = `
<DocumentFragment>
  <main
    class="MuiGrid-root MuiGrid-container sign-up css-1jrn2fe-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-1gbzzw4-MuiGrid-root"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1y8ugea"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-lguxed-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LockIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Sign Up
        </h1>
        <div
          class="sign-up-form"
        >
          <form
            action="#"
          >
            <div
              class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="email"
                id="email-label"
              >
                Email Address
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="email"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="email"
                  maxlength="120"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Email Address
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  maxlength="120"
                  name="password"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityOffIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7M2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2m4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="confirmPassword"
                id="confirmPassword-label"
              >
                Confirm password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="confirmPassword"
                  maxlength="120"
                  name="confirmPassword"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Confirm password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Sign Up
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Sing Up screen should submit signup form 1`] = `
<DocumentFragment>
  <main
    class="MuiGrid-root MuiGrid-container sign-up css-1jrn2fe-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-1gbzzw4-MuiGrid-root"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1y8ugea"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-lguxed-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LockIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Sign Up
        </h1>
        <div
          class="sign-up__success-message"
        >
          <div
            class="tick tick--lg"
          >
            <svg
              aria-hidden="true"
              class="checkmark"
              viewBox="0 0 52 52"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                class="checkmark__circle"
                cx="26"
                cy="26"
                fill="none"
                r="25"
              />
              <path
                class="checkmark__check"
                d="M14.1 27.2l7.1 7.2 16.7-16.8"
                fill="none"
              />
            </svg>
          </div>
          <h3
            class="text-center"
          >
            Success!
          </h3>
          <p>
            Thank you registered.
          </p>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
